File: Spring/Chapter7/MyBatisSpringJavaInit/src/main/java/pers/dynamicSQL/DynamicSQLDao.java
package pers.dynamicSQL;

import pers.init.entity.User;
import pers.init.pojo.UserPOJO;

import java.util.List;

public interface DynamicSQLDao {
    List<User> selectByIf(User user);
    List<User> selectByChoose(User user);
    List<User> selectByTrim(User user);
    List<User> selectByWhere(User user);
    int updateBySet(User user);
    List<User> selectByForeach(List<Integer> id);
    List<User> selectByBind(User user);
}

File: Spring/Chapter7/MyBatisSpringKotlinInit/src/main/kotlin/pers/dynamicSQL/DynamicSQLDao.kt
package pers.dynamicSQL

import pers.init.entity.User

interface DynamicSQLDao {
    fun selectByIf(user:User):List<User>
    fun selectByChoose(user:User):List<User>
    fun selectByTrim(user:User):List<User>
    fun selectByWhere(user:User):List<User>
    fun updateBySet(user:User):Int
    fun selectByForeach(id:List<Int>):List<User>
    fun selectByBind(user:User):List<User>
}